### PR TITLE
Added example for rendering images in subplots via matplotlib.

### PR DESCRIPTION
**Pull request by bdice**
## Description

Responding to a [user question on the Google Group](https://groups.google.com/d/msg/fresnel-users/sAmSog2WYmk/aDb_g85iBQAJ), I created an example showing how to use fresnel with matplotlib subplots.

## Checklist:
- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/fresnel-examples/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Fresnel Contributor Agreement**](https://github.com/glotzerlab/fresnel-examples/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of authors](https://github.com/glotzerlab/fresnel-examples/blob/master/AUTHORS.md).

### CODE REVIEW

**Comment by bdice** [comment]
@joaander Can you review this PR?

File: 02-Advanced-topics/04-Rendering-images-in-matplotlib.ipynb [hunk 1]
```diff
@@ -0,0 +1,192 @@
+{
```
Comment (joaander) [inline 02-Advanced-topics/04-Rendering-images-in-matplotlib.ipynb:1]: <p>`return np.asarray(out.buf)` should be `return out[:]`. `buf` should not be accessed directly by users.</p><p><br></p>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/fresnel-examples/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='4'/>

Comment (joaander) [inline 02-Advanced-topics/04-Rendering-images-in-matplotlib.ipynb:1]: <p>I would suggest a separate color for each shape. Consider black outlines for a more standard look.</p>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/fresnel-examples/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='6'/>

**Comment by bdice** [comment]
@joaander I updated the notebook with your suggestions. Thanks!

**Pull request merged**